store/store.go

```diff
@@ -308,29 +308,25 @@ func (s *Store[H]) DeleteRange(ctx context.Context, from, to uint64) error {
 		return fmt.Errorf("header/store: invalid range(%d,%d)", from, to)
 	}
 
-	last, err := s.GetByHeight(ctx, to-1)
-	if err != nil {
-		return fmt.Errorf("header/store: get by height: %w", err)
-	}
-
-	chunk := to - from
-	headers := make([]H, chunk)
-	for i := chunk - 1; i > 0; i-- {
-		headers[i], err = s.Get(ctx, last.LastHeader())
+	headers := make([]H, to-from)
+	for i := from; i < to; i++ {
+		var err error
+		headers[i], err = s.GetByHeight(ctx, i)
```

Comment:
sync: we don't need to load the whole header to delete, but only hash from the index

Reply:
Done! now much-much better